butane_core/src/migrations/adb.rs

```diff
@@ -461,3 +483,24 @@ fn diff_table(old: &ATable, new: &ATable) -> Vec<Operation> {
     }
     ops
 }
+
+#[cfg(test)]
+mod tests {
+    use std::collections::HashSet;
+
+    #[test]
+    /// Demostration that .difference() results are not stable.
+    fn show_unstable_migration_operation_order() {
```

Comment:
this test can be removed before merge - it doesnt test butane - it only shows using HashSet and difference/intersection is problematic.

I am now going to see whether using BTreeSet or `indexmap` crate provides another way of stablising the order without needing to alpha sort.

Reply:
Using BTreeSet doesnt offer any improvements - only the explicit `.sort()` can be removed.

using crate `indexmap` does allow "insertion order" of new tables and new columns, but it also requires that `ADB.tables` becomes a `indexmap::IndexMap` , and that means we need to enable indexmap feature `serde`.

I don't have a strong preference for insertion order vs alpha order.

db table explorers almost always show table lists and column lists in alpha order, so the order we insert them isnt likely to ever impact anyone. 

The only pro/con I can think of is that alpha order may make resolving merge conflicts a bit easier, if/when we ever get to helping the user resolve conflicting migrations during a rebase.

Reply:
in case it is useful for reviewing, here is what was here
```rust
        let old = ["a", "b"];
        let new = ["d", "b", "c"];
        let new_names: HashSet<&str> = new.iter().copied().collect();
        let old_names: HashSet<&str> = old.iter().copied().collect();
        let added_tables = new_names.difference(&old_names);
        let mut added_tables: Vec<&str> = added_tables.copied().collect();

        // Remove this line, and running this multiple times will result in a mix of passes and failures.
        added_tables.sort();

        assert_eq!(added_tables, vec!["c", "d"]);
```